src/iuse.cpp

```diff
@@ -4733,6 +4733,118 @@ int iuse::throwable_extinguisher_act(player *, item *it, bool, const tripoint &p
     it->active = false;
     return 0;
 }
+int iuse::capture_monster_act(player *p, item *it, bool, const tripoint& pos){
+  if(it->has_var("contained_name")){
+    try{
+        tripoint target;
+        if(g->is_empty(pos))
+        {
+            target=pos;//it's been activated somewhere where there isn't a player or monster, good.=
+        }else{
+            if(it->has_flag("PLACE_RANDOMLY")){
+                std::vector<tripoint> valid;
+                for(int x=p->posx()-1;x<=p->posx()+1;x++){
```

Comment:
Use `for( const tripoint &pt : g->m.points_in_radius( p->pos, 1 ) ) {` instead of double loop. It's clearer and easier to adjust.